src/coalesce.c

```diff
@@ -0,0 +1,76 @@
+#include "data.table.h"
+
+SEXP coalesce(SEXP x, SEXP values, SEXP n, SEXP count_na) {
+  int JJ = LENGTH(values), nprotect = 0, NN = asInteger(n), KK = asInteger(count_na);
+  
+  switch(TYPEOF(x)) {
+  case LGLSXP: {
+    int *base_ptr = LOGICAL(x);
+    
+    SEXP out = PROTECT(allocVector(LGLSXP, NN)); nprotect++;
+    int *out_ptr = LOGICAL(out);
+    
+    for (int i = 0; i < NN; i++) {
+      Rprintf("KK: %d\ti: %d\n", KK, i);
+      if (KK > 0 && base_ptr[i] == NA_LOGICAL) {
+        int j = 0;
+        while (j < JJ) {
```

Comment:
maybe create inline static functions for each data type, which can be cleanly referred from `while` loop?

Reply:
those loops are simplified now